portals/admin/source/src/app/components/Workflow/APIProductStateChange.jsx

```diff
@@ -0,0 +1,515 @@
+import Configurations from 'Config';
+import API from 'AppData/api';
+import Button from '@material-ui/core/Button';
+import * as dayjs from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import localizedFormat from 'dayjs/plugin/localizedFormat';
+import CheckIcon from '@material-ui/icons/Check';
+import ClearIcon from '@material-ui/icons/Clear';
+import Box from '@material-ui/core/Box';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import WarningBase from 'AppComponents/AdminPages/Addons/WarningBase';
+import { Alert as MUIAlert } from '@material-ui/lab';
+
+const useStyles = makeStyles((theme) => ({
+    searchInput: {
+        fontSize: theme.typography.fontSize,
+    },
+    block: {
+        display: theme.spacing(2),
+    },
+    contentWrapper: {
+        margin: '40px 16px',
+    },
+    approveButton: {
+        textDecoration: 'none',
+        backgroundColor: theme.palette.success.light,
+    },
+    rejectButton: {
+        textDecoration: 'none',
+        backgroundColor: theme.palette.error.light,
+    },
+}));
+
+/**
+ * Render a list
+ * @param {JSON} props props passed from parent
+ * @returns {JSX} Header AppBar components.
+ */
+function ListLabels() {
+    const intl = useIntl();
+    const [data, setData] = useState(null);
+    const restApi = new API();
+    const [isUpdating, setIsUpdating] = useState(null);
+    const [buttonValue, setButtonValue] = useState();
+    const [hasListPermission, setHasListPermission] = useState(true);
+    const [errorMessage, setError] = useState(null);
+
+    /**
+     * API call to get Detected Data
+     * @returns {Promise}.
+     */
+    function apiCall() {
+        return restApi
+            .workflowsGet('AM_API_PRODUCT_STATE')
+            .then((result) => {
+                const workflowlist = result.body.list.map((obj) => {
+                    return {
+                        description: obj.description,
+                        apiProduct: obj.properties.apiName,
+                        action: obj.properties.action,
+                        currentState: obj.properties.currentState,
+                        apiProvider: obj.properties.currentState,
+                        referenceId: obj.referenceId,
+                        createdTime: obj.createdTime,
+                        properties: obj.properties,
+                    };
+                });
+                return workflowlist;
+            })
+            .catch((error) => {
+                const { status } = error;
+                if (status === 401) {
+                    setHasListPermission(false);
+                } else {
+                    Alert.error(intl.formatMessage({
+                        id: 'Workflow.APIProductStateChange.apicall.has.errors',
+                        defaultMessage: 'Unable to get workflow pending requests for API Product State Change',
+                    }));
+                    throw (error);
```

Comment:
throw errors will send the error to a black hole. So best is to log the error. Refer bellow but don't go with the timeout approach, let's log the error.  console.log(error) instead of throw(error).

https://stackoverflow.com/questions/30715367/why-can-i-not-throw-inside-a-promise-catch-handler